doc: hubble configuration cleanup (#17522)
cilium-team/cilium: Documentation/concepts/observability/hubble-configuration.rst

Auto generated certificates via cert-manager
--------------------------------------------

> This method rely on `cert-manager <https://cert-manager.io/>`__ to generate the

Reviewer: ```suggestion
This method relies on `cert-manager <https://cert-manager.io/>`__ to generate the
```

> TLS certificates. cert-manager has becomes the de-facto way to manage TLS on

Reviewer 2: nit, 
```suggestion
TLS certificates. cert-manager has becomes the de facto way to manage TLS on
```

Kubernetes, and it has the following advantages compared to the previously
documented methods:

* Support multiple issuers (e.g. a custom CA,
  `Vault <https://www.vaultproject.io/>`__,
  `Let's Encrypt <https://letsencrypt.org/>`__,
  `Google's Certificate Authority Service <https://cloud.google.com/certificate-authority-service>`__,
  and more) allowing to choose the issuer fitting your organization's
  requirements.
* Manages certificates via a
  `CRD <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>`__
  which is easier to inspect with Kubernetes tools than PEM file.

**Installation steps**:

1. First, install `cert-manager <https://cert-manager.io/docs/installation/>`__
   and setup an `issuer <https://cert-manager.io/docs/configuration/>`_.
   Please make sure that your issuer is be able to create certificates under the
   ``cilium.io`` domain name.

> 2. Install/upgrade Cilium including the following Helm flags:

Reviewer 2: nit: I'm not so worried about this right here since I don't expect these instructions to change much, but you can use `#.` at the start of the lines to enumerate the numbers for you, and then as long as you keep the indentation correct, sphinx will automatically figure out the correct next `#.` number to place when rendering the pages. Can save on reworking the numbers in numbered lists that we expect to change over time.

Author: Marking as resolved as we don't enumerate the options anymore with the tabbed section.

Reviewer 2: :thinking: The numbered lists appear to still be here, the start of the line that this comment is based on has `2.`.

Author: My mistake, I thought this comment was referring to the workaround list we changed to tabbed below. Should be fixed now.

**Troubleshooting**:

While installing Cilium or cert-manager you may get the following error:

::

    Error: Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": dial tcp x.x.x.x:443: connect: connection refused

This happen when cert-manager's webhook (used to verify the ``Certificate`` CRD
resources) is not available. There are several ways to workaround this issue:

> * Option 1: install cert-manager CRDs before Cilium and cert-manager (see

Reviewer 2: We could consider changing these options into a tabbed section like we have for the install guide.

Author: Make sense!

**Troubleshooting**:

While installing Cilium or cert-manager you may get the following error:

::

    Error: Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": dial tcp x.x.x.x:443: connect: connection refused

This happen when cert-manager's webhook (used to verify the ``Certificate`` CRD

> resources) is not available. There are several ways to workaround this issue:

Reviewer 2: nit, visually users will get the hint but we can be a little bit more explicit here about what the user should do (in particular, do one, but don't do all). Also are these workarounds or are they they actual solution to this problem?

```suggestion
resources) is not available. There are several ways to resolve this issue.
Pick one of the options below:
```

.. tabs::

    .. group-tab:: Install CRDs first

        Install cert-manager CRDs before Cilium and cert-manager (see `cert-manager's documentation about installing CRDs with kubectl <https://cert-manager.io/docs/installation/helm/#option-1-installing-crds-with-kubectl>`__):

        .. code-block:: shell-session

            $ kubectl create -f cert-manager.crds.yaml

        Then install cert-manager, configure an issuer, and install Cilium.

>     .. group-tab:: upgrade Cilium

Reviewer 2: ```suggestion
    .. group-tab:: Upgrade Cilium
```

        Upgrade Cilium from an installation with TLS disabled:

        .. code-block:: shell-session

            $ helm install cilium cilium/cilium \
                --set hubble.tls.enabled=false \
                ...

        Then install cert-manager, configure an issuer, and upgrade Cilium enabling TLS:

        .. code-block:: shell-session

            $ helm install cilium cilium/cilium --set hubble.tls.enabled=true

>     .. group-tab:: disable webhook

Reviewer 2: ```suggestion
    .. group-tab:: Disable webhook
```

        Disable cert-manager validation (assuming Cilium is installed in the ``kube-system`` namespace):

        .. code-block:: shell-session

            $ kubectl label namespace kube-system cert-manager.io/disable-validation=true

        Then install Cilium, cert-manager, and configure an issuer.

>     .. group-tab:: hostns webhook

Reviewer 2: ```suggestion
    .. group-tab:: Host network webhook
```

**Troubleshooting**:

While installing Cilium or cert-manager you may get the following error:

::

    Error: Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": dial tcp x.x.x.x:443: connect: connection refused

> This happen when cert-manager's webhook (used to verify the ``Certificate`` CRD

Reviewer: ```suggestion
This happens when cert-manager's webhook (which is used to verify the ``Certificate``'s CRD
```
